### ISSUE TITLE
Implement psa_hash_suspend and psa_hash_resume

### ISSUE THREAD

**Issue by gilles-peskine-arm** [ARMmbed/mbedtls#3274]
The PSA hash API includes a way to serialize the state of a hash operation and deserialize it later, potentially on a different machine. The functions are called `psa_hash_suspend` and `psa_hash_resume`.

This is a relatively rare feature which should possibly be guarded by an ifdef.

**Comment by mpg** [comment]
Note: I don't think there are corresponding entry points in the driver API, should it be extended? (If so, we probably want a new issue to track that as well.)

**Comment by gilles-peskine-arm** [comment]
Yes, if we ever implement these functions, we should implement the corresponding driver interface as well.

**Comment by mpg** [comment]
Question: would it make sense to _define_ the entry points (in `docs/proposed/psa-crypto-driver-interface.md`) even if we don't _implement_ those functions? For the sake of other implementors.

**Comment by gilles-peskine-arm** [comment]
These functions exist because we had one internal customer, and I don't remember anyone else caring about them. So they are getting zero effort until someone actually cares.

**Comment by mpg** [comment]
Ok, makes sense. Thanks for clarifying that!
